sotodlib/preprocess/preprocess_util.py

```diff
@@ -406,64 +599,123 @@ def preproc_or_load_group(obs_id, configs, dets, logger=None,
+                    # pipeline for init config
+                    pipe_init = Pipeline(configs_init["process_pipe"], plot_dir=configs_init["plot_dir"], logger=logger)
+                    # pipeline for processing config
+                    pipe_proc = Pipeline(configs_proc["process_pipe"], plot_dir=configs_proc["plot_dir"], logger=logger)
+
+                    # tags from context proc
+                    tags_proc = np.array(context_proc.obsdb.get(aman.obs_info.obs_id, tags=True)['tags'])
+
+                    if "tags" in aman._fields:
+                        aman.move("tags", None)
+                    aman.wrap('tags', tags_proc)
+
+                    proc_aman, success = pipe_proc.run(aman)
+                    proc_aman.wrap('pcfg_ref', get_pcfg_check_aman(pipe_init))
+
+                    outputs_proc = save_group(obs_id, configs_proc, dets, context_proc, subdir='temp_proc')
+                    if overwrite or not os.path.exists(outputs_proc['temp_file']):
+                        logger.info(f"Saving data to {outputs_proc['temp_file']}:{outputs_proc['db_data']['dataset']}")
+                        proc_aman.save(outputs_proc['temp_file'], outputs_proc['db_data']['dataset'], overwrite)
+                    else:
+                        logger.info(f"{outputs_proc['temp_file']}:{outputs_proc['db_data']['dataset']} already exists.")
+
+                except Exception as e:
+                    error = f'Failed to load: {obs_id} {dets}'
+                    errmsg = f'{type(e)}: {e}'
+                    tb = ''.join(traceback.format_tb(e.__traceback__))
+                    logger.info(f"{error}\n{errmsg}\n{tb}")
+                    return error, [errmsg, tb], None
+                if success != 'end':
+                    # If a single group fails we don't log anywhere just mis an entry in the db.
+                    return success, [obs_id, dets], [obs_id, dets], None
+
+                return success, [obs_id, dets], outputs_proc, aman
     else:
+        # pipeline for init config
         logger.info(f"Generating new preproc db entry for {obs_id} {dets}")
-        pipe = Pipeline(configs["process_pipe"], plot_dir=configs["plot_dir"], logger=logger)
+        pipe_init = Pipeline(configs_init["process_pipe"], plot_dir=configs_init["plot_dir"], logger=logger)
         try:
-            aman = context.get_obs(obs_id, dets=dets)
-            tags = np.array(context.obsdb.get(aman.obs_info.obs_id, tags=True)['tags'])
+            aman = context_init.get_obs(obs_id, dets=dets)
+            tags = np.array(context_init.obsdb.get(aman.obs_info.obs_id, tags=True)['tags'])
             aman.wrap('tags', tags)
-            proc_aman, success = pipe.run(aman)
+            proc_aman, success = pipe_init.run(aman)
             aman.wrap('preprocess', proc_aman)
         except Exception as e:
             error = f'Failed to load: {obs_id} {dets}'
```

Comment:
These error messages should now include reference to whether this is error-ing out in the `init` or `proc` section to aid debugging where in this mess of nested if-else statements we're exiting.